src/pcapi/serialization/spec_tree.py

```diff
@@ -38,3 +41,35 @@ def _generate_spec(self) -> dict:
                     schema["paths"][path][method.lower()]["security"] = func.requires_authentication
 
         return schema
+
+
+class ExtendedResponse(SpectreeResponse):
+    def __init__(
+        self,
+        *codes,
+        code_descriptions=None,
+        **code_models,
+    ):
+        self.code_descriptions = code_descriptions or {}
+        super().__init__(*codes, **code_models)
+
+    def get_code_description(self, code):
+        return self.code_descriptions[code] if code in self.code_descriptions else DEFAULT_CODE_DESC[code]
+
+    def generate_spec(self):
+        """
+        extend the spec to add custom descriptions of error codes
+
+        :returns: JSON
+        """
+        responses = {}
+        for code in self.codes:
+            responses[parse_code(code)] = {"description": self.get_code_description(code)}
+
+        for code, model in self.code_models.items():
+            responses[parse_code(code)] = {
+                "description": self.get_code_description(code),
+                "content": {"application/json": {"schema": {"$ref": f"#/components/schemas/{model.__name__}"}}},
+            }
+
+        return responses
```

Comment:
(style / détail)

Ça reprend pas mal de code de la classe parente, et ça appelle 2 fois `get_code_description()`. Contre-proposition : appeler l'implémentation de la classe parente, et ne surcharger que ce qu'on veut :

```suggestion
        responses = super().generate_specs()
        all_codes = set(self.codes) | {code for code, model in self.code_models}
        for code in all_codes:
            description = self.code_descriptions.get(code)
            if description:  # override what the parent class set
                responses[parse_code(code)]["description"] = description
        return responses
```

... au passage, j'ai inliné `get_code_description()`. (Attention, je n'ai pas du tout testé mon code !)